pkg/assetfetcher/postgres.go

```diff
@@ -0,0 +1,52 @@
+package assetfetcher
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/asecurityteam/ipam-facade/pkg/domain"
+)
+
+const fetchByIPQuery = `SELECT host(d.ip) as ip, c.resource_owner as resource_owner,
+							c.business_unit as business_unit, text(s.network) as network,
+							s.location as location, d.id as device_id, s.id as subnet_id,
+							c.id as customer_id
+			            FROM devices d
+					  	RIGHT OUTER JOIN subnets s ON
+					  		d.subnet_id = s.id
+				  		AND d.ip = $1
+						INNER JOIN customers c ON s.customer_id = c.id
+						WHERE s.network >>= $1
+						ORDER BY d.id IS NOT NULL DESC, masklen(s.network) DESC
+						LIMIT 1;`
+
+// PostgresPhysicalAssetFetcher physical assets from a PostgreSQL database by IP address.
+type PostgresPhysicalAssetFetcher struct {
+	DB domain.SQLDB
+}
+
+// FetchPhysicalAsset queries the SQL DB for a physical asset by the given IP address.
+func (f *PostgresPhysicalAssetFetcher) FetchPhysicalAsset(ctx context.Context, ipAddress string) (domain.PhysicalAsset, error) {
+	var asset domain.PhysicalAsset
+	var ip sql.NullString
+	var deviceID sql.NullInt64
+	err := f.DB.Conn().QueryRowContext(ctx, fetchByIPQuery, ipAddress).Scan(
```

Comment:
task: You must close the connection in order for it to return to the pool.

> https://golang.org/pkg/database/sql/#Conn
> A Conn must call Close to return the connection to the database pool and may do so concurrently with a running query.

Reply:
This is `PostgresDB.Conn` which returns a `*sql.DB` pointer to a connection pool, not a `sql.DB.Conn`, which returns a `*sql.Conn` pointer to a single connection.

Reply:
No, I just didn't read the domain model closely enough.